Problem documents (#120)
abetterinternet/prio-documents: draft-pda-protocol.md

report shares are encrypted directly to the aggregators using HPKE {{!I-D.irtf-cfrg-hpke}}.

## Errors
Errors can be reported in PPM both at the HTTP layer and within

> challenge objects as defined in Section 8.  PPM servers can return

Reviewer: ```suggestion
challenge objects as defined in {{iana-considerations}}.  PPM servers can return
```

responses with an HTTP error response code (4XX or 5XX).  For
example, if the client submits a request using a method not allowed
in this document, then the server MAY return status code 405 (Method
Not Allowed).

When the server responds with an error status, it SHOULD provide
additional information using a problem document {{!RFC7807}}.  To
facilitate automatic response to errors, this document defines the
following standard tokens for use in the "type" field (within the
PPM URN namespace "urn:ietf:ppm:acme:error:"):

| Type                    | Description                                                                                  |
|:------------------------|:---------------------------------------------------------------------------------------------|

> | unrecognizedMessage     | The message type for a response was incorrect or the payload was malformed. 

Reviewer: ```suggestion
| unrecognizedMessage     | The message type for a response was incorrect or the payload was malformed. |
```

| unrecognizedTask        | An endpoint received a message with an unknown task ID |

> This list is not exhaustive.  The server MAY return errors whose

Reviewer: ```suggestion
This list is not exhaustive.  The server MAY return errors
```

[TODO: define how we want to structure this registry when the time comes]

## URN Sub-namespace for PPM (urn:ietf:params:ppm) {#ppm-urn-space}

The following value [will be/has been] registered in the "IETF URN Sub-
namespace for Registered Protocol Parameter Identifiers" registry,

> following the template in [RFC3553]:

Reviewer: ```suggestion
following the template in {{!RFC3553}}:
```
